File: 351003/Zhigimont/distcomp/publisher/src/main/java/com/bsuir/distcomp/controller/WriterController.java
package com.bsuir.distcomp.controller;

import com.bsuir.distcomp.dto.WriterRequestTo;
import com.bsuir.distcomp.dto.WriterResponseTo;
import com.bsuir.distcomp.service.WriterService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/writers")
@RequiredArgsConstructor
public class WriterController {

    private final WriterService service;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public WriterResponseTo create(@RequestBody @Valid WriterRequestTo dto) {
        return service.create(dto);
    }

    @GetMapping
    public List<WriterResponseTo> getAll() {
        return service.getAll();
    }

    @GetMapping("/{id}")
    public WriterResponseTo getById(@PathVariable Long id) {
        return service.getById(id);
    }

    @PutMapping("/{id}")
    public WriterResponseTo update(
            @PathVariable Long id,
            @RequestBody @Valid WriterRequestTo dto) {

        return service.update(id, dto);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
        service.delete(id);
    }
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/controller/WriterController.kt
package com.example.entitiesapp.controller

import com.example.entitiesapp.config.ApiConfig
import com.example.entitiesapp.dto.WriterRequestTo
import com.example.entitiesapp.dto.WriterResponseTo
import com.example.entitiesapp.exception.ValidationException
import com.example.entitiesapp.service.WriterService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["${ApiConfig.API_V1}/writers", "${ApiConfig.API_V2}/writers"])
class WriterController(
    private val service: WriterService
) {

    @GetMapping
    fun getAll(): ResponseEntity<List<WriterResponseTo>> =
        ResponseEntity.ok(service.getAll())

    @GetMapping("/{id}")
    fun getById(@PathVariable id: Long): ResponseEntity<WriterResponseTo> =
        ResponseEntity.ok(service.getById(id))

    @PostMapping
    fun create(@Valid @RequestBody dto: WriterRequestTo): ResponseEntity<WriterResponseTo> =
        ResponseEntity.status(HttpStatus.CREATED).body(service.create(dto))

    @PutMapping("/{id}")
    fun update(@PathVariable id: Long, @Valid @RequestBody dto: WriterRequestTo): ResponseEntity<WriterResponseTo> =
        ResponseEntity.ok(service.update(id, dto))

    @PutMapping
    fun updateFromBody(@Valid @RequestBody dto: WriterRequestTo): ResponseEntity<WriterResponseTo> {
        val id = dto.id ?: throw ValidationException("Id is required", 40000)
        return ResponseEntity.ok(service.update(id, dto))
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): ResponseEntity<Void> {
        service.delete(id)
        return ResponseEntity.noContent().build()
    }
}